internal/controller/api/v1alpha1/host_types.go

```diff
@@ -17,25 +17,188 @@ limitations under the License.
+}
+
+type HostInstallationEligibility struct {
+	Eligible           bool     `json:"eligible,omitempty"`
+	NotEligibleReasons []string `json:"notEligibleReasons"`
+}
+
+type HostIOPerf struct {
+	// 99th percentile of fsync duration in milliseconds
+	SyncDurationMilliseconds int64 `json:"syncDurationMilliseconds,omitempty"`
+}
+
+type HostDisk struct {
+	DriveType               string                      `json:"driveType,omitempty"`
+	Vendor                  string                      `json:"vendor,omitempty"`
+	Name                    string                      `json:"name,omitempty"`
+	Path                    string                      `json:"path,omitempty"`
+	Hctl                    string                      `json:"hctl,omitempty"`
+	ByPath                  string                      `json:"byPath,omitempty"`
+	Model                   string                      `json:"model,omitempty"`
+	Wwn                     string                      `json:"wwn,omitempty"`
+	Serial                  string                      `json:"serial,omitempty"`
+	SizeBytes               int64                       `json:"sizeBytes,omitempty"`
+	Bootable                bool                        `json:"bootable,omitempty"`
+	Smart                   string                      `json:"smart,omitempty"`
+	InstallationEligibility HostInstallationEligibility `json:"installationEligibility,omitempty"`
+	IoPerf                  HostIOPerf                  `json:"ioPerf,omitempty"`
+}
+
+type HostBoot struct {
+	CurrentBootMode string `json:"currentBootMode,omitempty"`
+	PxeInterface    string `json:"pxeInterface,omitempty"`
+}
+
+type HostSystemVendor struct {
+	SerialNumber string `json:"serialNumber,omitempty"`
+	ProductName  string `json:"productName,omitempty"`
+	Manufacturer string `json:"manufacturer,omitempty"`
+	Virtual      bool   `json:"virtual,omitempty"`
+}
+
+type HostInventory struct {
+	// Name in REST API: timestamp
+	ReportTime   *metav1.Time     `json:"reportTime,omitempty"`
+	Hostname     string           `json:"hostname,omitempty"`
+	BmcAddress   string           `json:"bmcAddress,omitempty"`
+	BmcV6address string           `json:"bmcV6Address,omitempty"`
+	Memory       HostMemory       `json:"memory,omitempty"`
+	Cpu          HostCPU          `json:"cpu,omitempty"`
+	Interfaces   []HostInterface  `json:"interfaces"`
+	Disks        []HostDisk       `json:"disks"`
+	Boot         HostBoot         `json:"boot,omitempty"`
+	SystemVendor HostSystemVendor `json:"systemVendor,omitempty"`
+}
 
 // HostSpec defines the desired state of Host
 type HostSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// The name of the Cluster CR
+	ClusterRef              *corev1.ObjectReference `json:"clusterRef"`
+	Role                    models.HostRole         `json:"role" protobuf:"bytes,1,opt,name=role,casttype=HostRole"`
```

Comment:
It would be useful to document here that the act of assigning a role will implicitly approve the host for provisioning.

Reply:
Ack. will add it in the next update

Reply:
The latest thinking is that we need a separate field to represent agent approval anyway, in part because we may utilize automation or inheritance of role based on labels or annotations from the BareMetalHost that booted the discovery ISO.

Reply:
What about self discovery? maybe we will not have bare metal host there?